src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12404,6 +12361,50 @@ private SyntaxList<SyntaxToken> ParseAnonymousFunctionModifiers()
             return result;
         }
 
+        private bool IsAnonymousFunctionAsyncModifier()
+        {
+            Debug.Assert(this.CurrentToken.ContextualKind == SyntaxKind.AsyncKeyword);
+
+            switch (this.PeekToken(1).Kind)
+            {
+                case SyntaxKind.OpenParenToken:
+                case SyntaxKind.IdentifierToken:
+                case SyntaxKind.StaticKeyword:
+                case SyntaxKind.RefKeyword:
+                case SyntaxKind.DelegateKeyword:
```

Comment:
Is this true? What if it's followed by a `*`?

Reply:
This method is just checking if `async` is in a location where it should be considered an `async` keyword in a lambda expression rather than an identifier. `async delegate*...` is a such a case.